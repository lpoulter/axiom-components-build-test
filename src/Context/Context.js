import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import Tip from '../Tip/Tip';
import './Context.css';

export default class Context extends Component {
  static propTypes = {
    arrowRef: PropTypes.func,
    children: PropTypes.node,
    className: PropTypes.string,
    color: PropTypes.oneOf(['success', 'warning', 'error', 'info', 'carbon', 'white']),
    maxHeight: PropTypes.string,
    position: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
    width: PropTypes.string,
  };

  static defaultProps = {
    maxHeight: '30rem',
    position: 'top',
    width: '14.5rem',
  };

  render() {
    const {
      className,
      arrowRef,
      children,
      maxHeight,
      position,
      width,
      color,
      ...rest
    } = this.props;

    const classes = classnames(className, 'ax-context', `ax-context--${position}`, {
      [`ax-context--${color}`]: color,
    });
    const content = () => <div className="ax-context__content" style={ { maxHeight } }>
      { children }
    </div>;
    return (
      <Base theme="day" { ...rest } className={ classes } style={ { width } }>
        { arrowRef ?
          <Tip arrowRef={ arrowRef } color={ color } direction={ position } size="small">
            { content() }
          </Tip> :
        content()
        }
      </Base>
    );
  }
}
