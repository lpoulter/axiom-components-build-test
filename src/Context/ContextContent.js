import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';

export default class ContextContent extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    color: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
    hasFullSeparator: PropTypes.bool,
    height: PropTypes.string,
    maxHeight: PropTypes.string,
    padding: PropTypes.oneOf(['none', 'tiny', 'small', 'medium', 'large']),
    paddingHorizontal: PropTypes.oneOf(['none', 'tiny', 'small', 'medium', 'large']),
    paddingVertical: PropTypes.oneOf(['none', 'tiny', 'small', 'medium', 'large']),
  };

  static defaultProps = {
    hasFullSeparator: false,
    padding: 'large',
  };

  render() {
    const {
      color,
      children,
      className,
      hasFullSeparator,
      height,
      maxHeight,
      padding,
      paddingHorizontal = padding,
      paddingVertical = padding,
      ...rest
    } = this.props;

    const classes = classnames(
      className,
      'ax-context-content',
      `ax-context-content--padding-horizontal-${paddingHorizontal}`,
      `ax-context-content--padding-vertical-${paddingVertical}`,
      {
        [`ax-context-content--${color}`]: color,
        'ax-context-content--full-separator': hasFullSeparator,
        'ax-context-content--scrollable': height || maxHeight,
      }
    );

    return (
      <Base { ...rest } className={ classes }>
        <div className="ax-context-content__scroll" style={ { height, maxHeight } }>
          { children }
        </div>
      </Base>
    );
  }
}
