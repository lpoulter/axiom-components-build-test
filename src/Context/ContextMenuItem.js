import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import CheckBox from '../Form/CheckBox';
import Icon from '../Icon/Icon';

export const contextMenuItemSelector = 'data-ax-context-menu-item';

export default class ContextMenuItem extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    icon: PropTypes.string,
    indeterminate: PropTypes.bool,
    multiSelect: PropTypes.bool,
    onClick: PropTypes.func,
    paddingVertical: PropTypes.oneOf(['small', 'medium']),
    selected: PropTypes.bool,
  };

  static defaultProps = {
    paddingVertical: 'medium',
  };

  render() {
    const {
      children,
      className,
      disabled,
      icon,
      indeterminate,
      onClick,
      multiSelect,
      selected,
      paddingVertical,
      ...rest
    } = this.props;

    const classes = classnames(
      className,
      'ax-context-menu__item',
      `ax-context-menu__item--padding-vertical-${paddingVertical}`, {
        'ax-context-menu__item--selected': selected,
      });

    return (
      <Base { ...rest } { ...{ [contextMenuItemSelector]: true } }
          Component="button"
          className={ classes }
          disabled={ disabled }
          onClick={ onClick }
          textStrong={ selected }>
        { multiSelect && (
          <div className="ax-context-menu__item-checkbox">
            <CheckBox
                checked={ selected }
                disabled={ disabled }
                indeterminate={ indeterminate }
                onChange={ onClick }
                onClick={ e => e.stopPropagation() }/>
          </div>
        ) }

        <div className="ax-context-menu__item-content">
          { children }
        </div>

        { !multiSelect && selected !== undefined && (
          <div className="ax-context-menu__item-icon">
            <Icon cloak={ !selected } name="tick" />
          </div>
        ) }

        { icon !== undefined && (
          <div className="ax-context-menu__item-icon">
            <Icon name={ icon } />
          </div>
        ) }
      </Base>
    );
  }
}
