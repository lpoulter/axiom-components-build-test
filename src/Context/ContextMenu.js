import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import ContextContent from './ContextContent';

export default class ContextMenu extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    paddingVertical: PropTypes.oneOf(['none', 'medium']),
  };

  static defaultProps = {
    hasFullSeparator: false,
    paddingVertical: 'medium',
  };

  render() {
    const { children, className, paddingVertical, ...rest } = this.props;

    const classes = classnames(
      className,
      'ax-context-menu',
      `ax-context-menu--padding-vertical-${paddingVertical}`,
    );

    return (
      <ContextContent { ...rest } padding="none">
        <div className={ classes }>
          { children }
        </div>
      </ContextContent>
    );
  }
}
