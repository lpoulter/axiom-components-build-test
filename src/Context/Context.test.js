import React from 'react';
import renderer from 'react-test-renderer';
import Context from './Context';

const getComponent = (props = {}, context = {}) =>
  renderer.create(
    <Context { ...props } { ...context }>
      Lorem ipsum
    </Context>
  );

describe('Context', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with arrowRef', () => {
    const component = getComponent({ arrowRef: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders custom maxHeight', () => {
    const component = getComponent({ maxHeight: '20rem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders custom width', () => {
    const component = getComponent({ width: '20rem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with position', () => {
    ['top', 'bottom', 'right', 'left'].forEach((position) => {
      it(position, () => {
        const component = getComponent({ position });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with theme', () => {
    ['day', 'night'].forEach((theme) => {
      it(theme, () => {
        const component = getComponent({ theme });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with "type" context', () => {
    ['success', 'warning', 'error', 'info'].forEach(type => {
      it(type, () => {
        const component = getComponent({}, { type });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
