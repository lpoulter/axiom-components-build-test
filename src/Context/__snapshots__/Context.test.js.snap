// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context renders custom maxHeight 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "20rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders custom width 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "20rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with "type" context error 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
  type="error"
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with "type" context info 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
  type="info"
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with "type" context success 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
  type="success"
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with "type" context warning 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
  type="warning"
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with arrowRef 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-tip--top ax-tip--shadow"
  >
    <span
      className="ax-tip__arrow ax-tip__arrow--small"
      style={Object {}}
    />
    <div
      className="ax-tip__content"
    >
      <div
        className="ax-context__content"
        style={
          Object {
            "maxHeight": "30rem",
          }
        }
      >
        Lorem ipsum
      </div>
    </div>
  </div>
</div>
`;

exports[`Context renders with defaultProps 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with position bottom 1`] = `
<div
  className="ax-context ax-context--bottom ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with position left 1`] = `
<div
  className="ax-context ax-context--left ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with position right 1`] = `
<div
  className="ax-context ax-context--right ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with position top 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with theme day 1`] = `
<div
  className="ax-context ax-context--top ax-theme--day"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Context renders with theme night 1`] = `
<div
  className="ax-context ax-context--top ax-theme--night"
  style={
    Object {
      "width": "14.5rem",
    }
  }
>
  <div
    className="ax-context__content"
    style={
      Object {
        "maxHeight": "30rem",
      }
    }
  >
    Lorem ipsum
  </div>
</div>
`;
